crates/ra_ide_api/src/matching_brace.rs

```diff
@@ -14,9 +17,15 @@ pub fn matching_brace(file: &SourceFile, offset: TextUnit) -> Option<TextUnit> {
         })
         .next()?;
     let parent = brace_node.parent();
-    let matching_kind = BRACES[brace_idx ^ 1];
+    let brace_pos = brace_idx ^ 1;
+    let matching_kind = BRACES[brace_pos];
     let matching_node = parent.children_with_tokens().find(|node| node.kind() == matching_kind)?;
-    Some(matching_node.text_range().start())
+
+    match (brace_pos % 2) as i32 {
```

Comment:
```suggestion
    match brace_pos & 2 {
```

And then change `L_BRACE` and `R_BRACE` to the same type as `brace_pos`.

Reply:
Did you mean `& 1`? I think `% 2` is more clear, though.

Reply:
Yes. `% 2` is fine too. The `as i32` part should go though.